[Ho Shang Ji Jason] iP

Jason-Hp opened the pull request.

Jason-Hp committed:
Add Increment: Level-0

Jason-Hp committed:
Import Scanner and refactor Mike.java

Jason-Hp committed:
Add echo feature for Mike.java

Jason-Hp committed:
Add ability to store user's input(s) and display them as a list

Jason-Hp committed:
Add Task.java

Jason-Hp committed:
Complete Task class with basic attributes and methods

Jason-Hp committed:
Add Mark as Done feature and refactored Main.java

Jason-Hp committed:
Add unmark feature

Jason-Hp committed:
Reformat code to comply with coding standard

Jason-Hp committed:
Add new field symbol to Task class

Jason-Hp committed:
Create Todo, Deadline and Event classes that inherits from Task

Jason-Hp committed:
Update Deadline, Event and Task classes

Jason-Hp committed:
Update Mike.java to support Todo, Deadline and Event classes

Update existing code so to accomodate new classes that are instances
or subclasses of the previously used Task class. Main features are still similar.

Jason-Hp committed:
Refactor Mike.java

Extract constants like Line separator and White/Blank spacing
Extract constant of maximum number of items (Avoid magic number)
Extract method of performing marking/unmarking

Jason-Hp committed:
Refactor Mike.java and Update printing of ERROR

Extract constants and remove magic numbers like the number of characters of
todo and a space, deadline and a space, and, event and a space into CONSTANTS.

Update ERROR case to align to the standard way of printing to the CLI, by printing
out Line separator and include correct indents(Spaces).

HuaZhenting commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+    protected String from;
+    protected String to;
+
+    public Event(String description,String from,String to){
```
Please be noted that there should be a space after ",".

HuaZhenting commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+
+    public Event(String description,String from,String to){
+        super(description + " (from: "+from+" to: "+to+")");
+        this.from=from;
```
There should be spaces around the = operator for better readability.

HuaZhenting commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,169 @@
+    }
+
+    public static String itemToString(Task task){
+        return
```
Note that something should follow return in this line

HuaZhenting commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,169 @@
+            case "list":
+                System.out.println(LINE_SEPARATOR+"\n" +
+                        SPACES+"Here are the tasks in your list:");
+                
```
Please note that there should not be spare lines.

KashfyZul commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task{
+
```
I like how you inherited the attributes of Task into Deadline

KashfyZul commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,169 @@
+            "`------'`------'`------'`------'";
+    public static final String LINE_SEPARATOR = "    ______________________________"+
+            "______________________________";
+
```
I like how you defined the constants and even the LINE_SEPARATOR, very easy to read

KashfyZul commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,169 @@
+        String todo = userInput.substring(TODO_AND_SPACE);
+        return new Todo(todo);
+    }
+
```
Nice parser, clever use of substring with TODO_AND_SPACE

KashfyZul commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,169 @@
+                userInput = scanner.nextLine();
+                continue;
+            }
+
```
I like your use of case statements, very easy to follow

KashfyZul commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,169 @@
+                System.out.println(LINE_SEPARATOR+"\n" +
+                        "     OK, I've marked this task as not done yet:");
+            }
+
```
Perhaps you could include the case whereby the task is already marked as done but the user inputs to mark it as done again

KashfyZul reviewed (commented):
LGTM, overall very easy to read and understand

Jason-Hp committed:
Create RandomException class

Jason-Hp committed:
Create EmptyException class

Jason-Hp committed:
Update Mike.java to hanlde exceptions

Add try and catch blocks, and made existing methods throw exceptions if
and when required like random commands or empty descriptions.

Jason-Hp committed:
Merge branch 'branch-Level-5'

Jason-Hp committed:
Organize project structure: Create packages and move files

Jason-Hp committed:
Merge branch 'branch-A-Packages'

Jason-Hp committed:
Update code to use dynamic ArrayList instead

Jason-Hp committed:
Implement delete feature

Jason-Hp committed:
Add file handler package and data

Jason-Hp committed:
Update Mike.java to handle files

Jason-Hp committed:
Merge branch 'branch-Level-6'

Jason-Hp committed:
Resolve merge conflicts with branch-Level-7

Jason-Hp committed:
Update code to not use absolute path and bug fixes

Uses relative path instead, additonally, fixed a few bugs that causes event and deadline to be stored as todo object.

Jason-Hp committed:
Fix bug that prints unnecessary information

Jason-Hp committed:
Add delete for file/text as well

Jason-Hp committed:
Refactor code to extract code as classes

Extract ArrayList<Task> list, all methods to update the list, all parsers of user input, interaction betwwen CLI and user.

Jason-Hp committed:
Refactor code to include new extracted classes

Jason-Hp committed:
Fix bug relating to marking and unmarking tasks

Jason-Hp committed:
Add assertion into code

Jason-Hp committed:
Clean up code to look more readable

Jason-Hp committed:
Refactor code to include Stream

Jason-Hp committed:
Merge pull request #1 from Jason-Hp/branch-A-Assertions

Add assertion into code

Jason-Hp committed:
Merge branch 'master' into branch-A-CodeQuality

Jason-Hp committed:
Merge branch 'master' into branch-A-Streams

Jason-Hp committed:
Merge pull request #2 from Jason-Hp/branch-A-CodeQuality

Clean up code to look more readable

Jason-Hp committed:
Merge branch 'master' into branch-A-Streams

Jason-Hp committed:
Merge pull request #3 from Jason-Hp/branch-A-Streams

Refactor code to include Stream

Jason-Hp committed:
Refactor code for more readability

Jason-Hp committed:
Add find feature into code

Jason-Hp committed:
Merge branch 'branch-Level-9'

Jason-Hp committed:
Add JavaDoc comments to most non-private classes/methods

Jason-Hp committed:
Merge branch 'branch-A-JavaDoc'

Jason-Hp committed:
Update README.md with a user guide

Jason-Hp committed:
Build JAR file

Jason-Hp committed:
Add directory and file creation if don't exist

Jason-Hp committed:
Update README.md

Jason-Hp committed:
Update README.md

aditichadha1310 commented on src/main/java/Mike.java:
```diff
@@ -0,0 +1,156 @@
+     * Runs or starts the Mike object for interaction between
+     * user and Mike, using the CLI for any user interactions
+     */
+    public void run(){
```
Avoid lengthy methods. try to keep the method length to about 30 lines of code.

aditichadha1310 commented on src/main/java/filehandler/FileReader.java:
```diff
@@ -0,0 +1,78 @@
+                String time = information[3];
+                String[] fromAndTo = time.split(" to ");
+                Event eventToBeAdded = new Event(taskDescription, fromAndTo[0].split("from ")[1], fromAndTo[1]);
+                if (isMarked.equals("1")) {
```
Avoid usage of magic strings and numbers in the code.

aditichadha1310 reviewed (commented):
overall, good work!
